src/async_wrap.cc

```diff
@@ -226,7 +246,27 @@ static PromiseWrap* extractPromiseWrap(Local<Promise> promise) {
   return obj->IsObject() ? Unwrap<PromiseWrap>(obj.As<Object>()) : nullptr;
 }
 
-static void PromiseHook(PromiseHookType type, Local<Promise> promise,
+// Simplified JavaScript hook fast-path for when there is no destroy hook
+static void FastPromiseHook(PromiseHookType type, Local<Promise> promise,
+                        Local<Value> parent) {
+  Local<Context> context = promise->CreationContext();
+
+  Environment* env = Environment::GetCurrent(context);
+  if (env == nullptr) return;
+  TraceEventScope trace_scope(TRACING_CATEGORY_NODE1(environment),
+                              "EnvPromiseHook", env);
+
+  Local<Value> argv[] = {
+    env->isolate_data()->promise_hook_type(static_cast<int>(type)),
+    promise,
+    parent
+  };
+
+  Local<Function> promise_hook = env->promise_hook_handler();
+  USE(promise_hook->Call(context, Undefined(env->isolate()), 3, argv));
```

Comment:
What if there is an exception? Should this have a `TryCatchScope try_catch(env, TryCatchScope::CatchMode::kFatal);` line, like the other JS entry points here?

Reply:
Probably, yes.